signing_service/signing_service/signing_service.py

```diff
@@ -138,7 +138,11 @@ def run(self) -> None:
                 except (socket.error, socket.timeout) as exception:
                     logger.error(f'Socket error occurred: {exception}')
 
-                    self._attempt_reconnection()
+                    try:
+                        self._attempt_reconnection()
+                    except SigningServiceMaximumReconnectionAttemptsExceeded:
+                        logger.error('Maximum number of reconnections exceeded.')
+                        break
```

Comment:
This break will close the `SigningService` - am I right? Is it our intention?

Reply:
Yes. Also, it was working exactly like this before this change.